game_app/games/tic_tac_toe/tic_tac_toe_component.py

```diff
@@ -88,60 +92,53 @@ def receiveMessage(self, msg, sender):
 
         elif isinstance(msg, MatchMakerUninitializedMsg):
-            self.log(f"Received MatchMakerUninitializedMsg message from {sender}")
             self.log("Can't join server because MatchMaker hasn't ben initialized")
             raise MatchMakerUninitializedError
 
         elif isinstance(msg, InvalidPlayerMsg):
-            self.log(f"Received InvalidPlayerMsg message from {sender}")
             self.log("Invalid player sent during joining server")
             raise InvalidPlayer
 
         elif isinstance(msg, JoinAcknowledgementsMsg):
-            self.log(f"Received JoinAcknowledgementsMsg message from {sender}")
             self.log("Successfully joined server!")
 
         # Main Game loop
         elif isinstance(msg, YourTurnMsg):
-            self.log(f"Received YourTurnMsg message from {sender}")
             state_changed_event = {"type": UserEventTypes.STATE_CHANGED.value, "new_game_state": msg.state}
             turn_changed_event = {"type": UserEventTypes.TURN_CHANGED.value, "new_turn": TurnState.NEW_YOUR_TURN}
             self._events_to_post += [state_changed_event, turn_changed_event]
 
         elif isinstance(msg, MoveMsg):
-            self.log(f"Received MoveMsg message from {sender}")
             self.send(self.game_manager_addr, TakeActionMsg(msg.action))
-            self.log(f"Sent TakeActionMsg message to GameManager {self.game_manager_addr}")
-
-        elif isinstance(msg, RewardMsg):
-            pass
 
         elif isinstance(msg, GameOverMsg):
-            self.log(f"Received GameOverMsg message from {sender}")
             game_over_event = {"type": UserEventTypes.GAME_OVER.value, "new_winnings": msg.winnings}
             self._events_to_post.append(game_over_event)
 
         elif isinstance(msg, StateUpdateMsg):
-            self.log(f"Received StateUpdateMsg message from {sender}")
             state_changed_event = {"type": UserEventTypes.STATE_CHANGED.value, "new_game_state": msg.state}
             self._events_to_post += [state_changed_event]
 
         # Exiting
         elif isinstance(msg, ActorExitRequest):
-            self.log(f"Received ActorExitRequest message from {sender}")
-
+            self.log(f"Exiting")
         else:
-            self.log(f"Received unexpected message {msg} of type {msg} from {sender}")
             raise UnexpectedMessageError(msg)
 
-    def log(self, text):
+    def log(self, text, logging_level=LoggingLevel.GAME_EVENTS):
+        if not LOGGING:
+            return
         if self.logger_addr is not None:
-            self.send(self.logger_addr, LogMsg(text, f"GUI client:{self.player}"))
+            super().send(self.logger_addr, LogMsg(text, f"GUI client:{self.player}", logging_level))
+
+    def send(self, target_address, message):
+        super().send(target_address, message)
+        if not isinstance(message, EventsToPostMsg):
+            self.log(f"Sent {message} to {target_address}", LoggingLevel.PLATFORM_COMMUNICATION_MESSAGES)
```

Comment:
wyrzucilbym te funkcje na gore klasy, pod __init__ zeby bylo wiadomo skad sie biora logi czytajac po kolei, bo to takie troche sneaky podejscie, ale jak nie uwazasz tego za dobry pomysl to olej

Reply:
Dobry pomysł, przeniosę